### ISSUE TITLE
Cannot add OffMeshConnections to NavMeshBuilder

### ISSUE THREAD

**Issue by PixelRetro** [Robmaister/SharpNav#53]
Adding OffMeshConnections to the NavMeshBuilder results in an NullReferenceException, at [Line 273](https://github.com/Robmaister/SharpNav/blob/master/Source/SharpNav/NavMeshBuilder.cs#L273), because there is no instance of the Poly class at this position in the navPoly array.
Was Poly a struct some time ago?

**Comment by PixelRetro** [comment]
The same happens when storing the OffMeshConnections at [Line 361](https://github.com/Robmaister/SharpNav/blob/master/Source/SharpNav/NavMeshBuilder.cs#L361).

**Comment by PixelRetro** [comment]
Pull request: #55

**Comment by Robmaister** [comment]
So yeah, they used to be structs a while ago (when it was more of a bare port of Recast) and when I started refactoring it to look more like other C# libraries I changed them to classes and missed a few bits.

I'm still in the process of updating and cleaning up all the pathfinding API. Thanks for finding this bug and submitting the pull request.

Also note that I am pretty busy so I won't have too much bandwidth to hunt and fix a lot of small bugs, I should still be pretty responsive on GitHub if you have questions, though.
